akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -89,6 +89,11 @@ private[remote] class Encoder(
         val outboundEnvelope = grab(in)
         val envelope = bufferPool.acquire()
 
+        headerBuilder.resetMessageFields()
+        // don't use outbound compression for ArteryMessage, e.g. handshake messages must get through
+        // without depending on compression tables being in sync when systems are restarted
+        headerBuilder.useOutboundCompression(!outboundEnvelope.message.isInstanceOf[ArteryMessage])
```

Comment:
this is a good idea. It should not have any performance impact. If it does, then we can revisit some messages.